File: roboplan_examples/cpp/example_scene.cpp
#include <filesystem>
#include <iostream>
#include <vector>

#include <roboplan/core/scene.hpp>
#include <roboplan_example_models/resources.hpp>

using namespace roboplan;

int main(int /*argc*/, char* /*argv*/[]) {
  // Set up the scene
  const auto share_prefix = example_models::get_package_share_dir();
  const auto model_prefix = share_prefix / "roboplan_example_models" / "models";
  const auto urdf_path = model_prefix / "ur_robot_model" / "ur5_gripper.urdf";
  const auto srdf_path = model_prefix / "ur_robot_model" / "ur5_gripper.srdf";
  const std::vector<std::filesystem::path> package_paths = {share_prefix};

  auto scene = Scene("example_scene", urdf_path, srdf_path, package_paths);
  std::cout << scene;

  // Generate a random state
  auto random_positions = scene.randomPositions();
  std::cout << "Random positions:\n" << random_positions.transpose() << "\n";
  scene.setRngSeed(1234);
  random_positions = scene.randomPositions();
  std::cout << "Random positions (seeded):\n" << random_positions.transpose() << "\n";

  return 0;
}

File: roboplan_examples/python/example_scene.py
#!/usr/bin/env python3

import numpy as np

from common import get_model_data
from roboplan.core import JointConfiguration, Scene
from roboplan.example_models import get_package_share_dir


if __name__ == "__main__":

    jc = JointConfiguration(
        ["joint_1", "joint_2", "joint_3"],
        np.array([0.1, 0.2, 0.3]),
    )
    print(f"Config names: {jc.joint_names}")
    print(f"Config positions: {jc.positions}")
    print("")

    model = "ur5"
    model_data = get_model_data()[model]
    package_paths = [get_package_share_dir()]

    scene = Scene(
        "test_scene", model_data.urdf_path, model_data.srdf_path, package_paths
    )
    print(scene)
